### ISSUE TITLE
Password field uses new-password instead of current-password

### ISSUE THREAD

**Issue by skookie000** [ExpressionEngine/ExpressionEngine#209]
** Problem **
A password conformation fields have the HTML attribute "new-password" instead of probably "current-password."  Browsers won't use saved password, and some [Safari] try to auto-fill new hash => annoyance.
Example:
/cp/members/profile/email&id=1

**Environment Details:**
 - Version: 5.2.2
 - PHP Version 7.2

**Possible Solution**
field.php 86 <input type="password" name="<?=$field_name?>" value="<?=$value?>" autocomplete="new-password"<?=$attrs?> class="<?=$class?>">

https://developer.apple.com/documentation/security/password_autofill/enabling_password_autofill_on_an_html_input_element

**Comment by bmackenty** [comment]
Hi! 

Newbie contributor here.

For username / password change page we have three fields in the password section: 

1. new password (autocomplete = "new-password", name="password")
2. confirm new password (autocomplete = "new-password", name = "confirm_password")
3. old password (autocomplete = "new-password", name="verify_password")

You want the old password to have autocomplete "current-password" instead of "new-password".  Is that a correct understanding? 

Currently, in:  `system/ee/EllisLab/ExpressionEngine/View/_shared/form/field.php`

The new-password autocomplete text is generated via a switch / case, so maybe we would need to add a conditional that looks at the name and if the name equals "verify_password" set the autocomplete to current_password. It looks like "verify_password" is hardcoded in `system/ee/EllisLab/ExpressionEngine/Controller/Members/Profile/Profile.php` around line 303. 

Maybe something like this: 

old code: 
`
case 'password': ?>
	<input type="password" name="<?=$field_name?>" value="<?=$value?>" autocomplete="new-password"<?=$attrs?> class="<?=$class?>">
`
proposed code: 
`
case 'password': ?>
 if ($field['name'] == 'verify_password') {
	<input type="password" name="<?=$field_name?>" value="<?=$value?>" autocomplete="current-password"<?=$attrs?> class="<?=$class?>"> 
} else {
<input type="password" name="<?=$field_name?>" value="<?=$value?>" autocomplete="current-password"<?=$attrs?> class="<?=$class?>">
}
`

**Comment by skookie000** [comment]
Correct, though I missed the password reset field shares the same logic.  That's not ideal.  Ideally new password fields have 'new-password' and current password fields (like to make admin changes) have 'current-password'.

/cp/members/profile/email&id=1 Would need 'current-password' in the "Enter Your Password" field.
/cp/members/profile/auth&id=1 Would need both.
/cp/members/profile/login&id=1234 Would need 'current-password'
/cp/members/create Would need both.
...etc.

Alternatively if they can't be separated it might be better to remove autocomplete.

**Comment by bmackenty** [comment]
So it looks like we might need to create a context for each of the different places we ask for a new password or to confirm a password so we can then decide if a field should be new or verify.  I think an inventory of every page we use this would be helpful. 

If addon's reference field types for password stuff (I'm thinking user, maybe) I think this issue might be a little over my head. I'll ping on of the EE guys to poke into this issue :-)

**Comment by derekjones** [comment]
Sorry, dropping in without a lot of context. There are two different confirmation situations:

1. A user needing to take an action that should confirm their password
2. An admin taking an action that should require then to reconfirm (or have done so in the last 15 minutes / aka sudo power)

Where exactly are the problem areas in the control panel with that? I use Safari and 1Password, so that may be disguising the issue for me.

**Comment by intoeetive** [comment]
@bmackenty I have checked your proposed PR and I think the changes you suggest are completely legit.

Except that since the only change is in `new`/`current` part, might make sense to use ternary operator just in that particular part, keeping the code DRY

Would you mind making this change for EE6 and submitting pull request against `6.dev`?

**Comment by bmackenty** [comment]
oof!

I'm deep into projects which are pulling my time. I'm afraid I won't be
able to commit to this fix until end of summer. :-(

**Comment by intoeetive** [comment]
No worries, I'll do that myself! Thank you for responding
